packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/private/helpers.ts

```diff
@@ -64,3 +64,26 @@ export function createLoggingOptions(scope: Construct, props: DestinationLogging
   }
   return undefined;
 }
+
+export function createBufferingHints(
+  bufferingInterval?: cdk.Duration,
+  bufferingSize?: cdk.Size,
+):firehose.CfnDeliveryStream.BufferingHintsProperty | undefined {
+  if (bufferingInterval && bufferingSize) {
+    if (bufferingInterval.toSeconds() < 60 || bufferingInterval.toSeconds() > 900) {
+      throw new Error('Buffering interval must be between 60 and 900 seconds');
+    }
+    if (bufferingSize.toMebibytes() < 1 || bufferingSize.toMebibytes() > 128) {
+      throw new Error('Buffering size must be between 1 and 128 MBs');
+    }
+    return {
+      intervalInSeconds: bufferingInterval.toSeconds(),
+      sizeInMBs: bufferingSize.toMebibytes(),
+    };
+  } else if (!bufferingInterval && bufferingSize) {
+    throw new Error('If bufferingSize is specified, bufferingInterval must also be specified');
+  } else if (bufferingInterval && !bufferingSize) {
+    throw new Error('If bufferingInterval is specified, bufferingSize must also be specified');
+  }
```

Comment:
Can we re-structure the code so that the error checking is first, and the return is last? Just makes it more natural to read a function in that order.